pkg/router/istio.go

```diff
@@ -440,15 +440,26 @@ func (ir *IstioRouter) Finalize(canary *flaggerv1.Canary) error {
 
 // mergeMatchConditions appends the URI match rules to canary conditions
 func mergeMatchConditions(canary, defaults []istiov1alpha3.HTTPMatchRequest) []istiov1alpha3.HTTPMatchRequest {
-	for i := range canary {
+	if len(defaults) == 0 {
+		return canary
+	}
+
+	merged := make([]istiov1alpha3.HTTPMatchRequest, len(canary)*len(defaults))
```

Comment:
Shouldn't this be `len(canary)+len(defaults)`?

Reply:
@stefanprodan  Thank you for your reviewing.
My intention is that `len(canary)*len(defaults)`  is correct.
Because I think it should be expected to work as follows.

**Canary Definitions**<br>
```yaml
  service:
    # accepts /podinfo or /_aliases/pi.
    match:
      - uri.prefix: /podinfo            # a)
      - uri.prefix: /_aliases/pi        # b)
  ...
  analysis:
    # if contained specific user-agent or cookie, route to canary
    match:
      - headers:                        # c)
          user-agent:
            exact: "Developers-Mode"
      - headers:                        # d)
          cookie:
            regex: "^(.*?;)?(type=insider)(;.*)?$"
```


**Expected to generated Virtual Service of Canary**<br>
```yaml
  http:
    match:
      - uri.prefix: /podinfo       # a) + c)
        headers:
          user-agent:
            exact: "Developers-Mode"
      - uri.prefix: /podinfo       # a) + d)
        headers:
          cookie:
            regex: "^(.*?;)?(type=insider)(;.*)?$"
      - uri.prefix: /_aliases/pi   # b) + c)
        headers:
          user-agent:
            exact: "Developers-Mode"
      - uri.prefix: /_aliases/pi   # b) + d)
        headers:
          cookie:
            regex: "^(.*?;)?(type=insider)(;.*)?$"
```
I think I have to create http.match by a service.match rule for each analysis.match.
I think `len(canary)*len(defaults)` is correct.

However, if the logic is different from what you originally expected, Please let me know correct logic.
I will fix this and the merge logic as well.

Reply:
Ah you are right, this makes total sense sorry for the noise.